code/modules/mob/living/simple_animal/hostile/hivebot.dm

```diff
@@ -65,43 +63,48 @@
 	status_flags = 0
 	anchored = 1
 	stop_automated_movement = 1
-	var/bot_type = "norm"
+	var/bot_type = 1 // type of bot, 1 is normal, 2 is ranged, 3 is rapid rnaged.
 	var/bot_amt = 10
 	var/spawn_delay = 600
-	var/turn_on = 0
-	var/auto_spawn = 1
-	proc
-		warpbots()
-
-
-	New()
-		..()
-		var/datum/effect/effect/system/smoke_spread/smoke = new /datum/effect/effect/system/smoke_spread()
-		smoke.set_up(5, 0, src.loc)
-		smoke.start()
-		visible_message("<span class='danger'>The [src] warps in!</span>")
-		playsound(src.loc, 'sound/effects/EMPulse.ogg', 25, 1)
-
-	warpbots()
-		icon_state = "def_radar"
-		visible_message("<span class='warning'>The [src] turns on!</span>")
-		while(bot_amt > 0)
-			bot_amt--
-			switch(bot_type)
-				if("norm")
-					new /mob/living/simple_animal/hostile/hivebot(get_turf(src))
-				if("range")
-					new /mob/living/simple_animal/hostile/hivebot/range(get_turf(src))
-				if("rapid")
-					new /mob/living/simple_animal/hostile/hivebot/rapid(get_turf(src))
-		spawn(100)
-			qdel(src)
-		return
-
-
-	Life()
-		..()
-		if(stat == 0)
-			if(prob(2))//Might be a bit low, will mess with it likely
-				warpbots()
+
+/mob/living/simple_animal/hostile/hivebot/tele/New()

```

Comment:
Initialize > New

Reply:
I only translated code into absolute pathing. I believe they use New as this behavior is intended to occur during beacon spawn.